### PR TITLE
Fix/master/sitemap

### LINKED ISSUE #496: TxNewsSitemapGenerator should respect news type
Pages with the types "internal link" or "external link" will get the same paths as regular news. This is different from the behavior of LinkViewHelper which will `getInternalurl()` or `getExternalurl()` as a link parameter.

The proposed solution is as follows: In `getNewsItemUrl()` insert
```php
if ($newsRow['type'] == 1 && !empty($newsRow['internalurl'])) {
    $conf['additionalParams'] = $additionalParams;
    $conf['parameter'] = $newsRow['internalurl'];
}
if ($newsRow['type'] == 2 && !empty($newsRow['externalurl'])) {
    $conf['additionalParams'] = $additionalParams;
    $conf['parameter'] = $newsRow['externalurl'];
}
$link = htmlspecialchars($this->cObj->typoLink('', $conf));
```
The last line is from the original function right before the final `return` statement.

This has been successfully tested on TYPO3 8.7.8 with news 6.1.1 and internal news. We don't use external urls.

### PR DESCRIPTION
**Pull request by 0bserver**
This fix resolves #496.

### CODE REVIEW

File: Classes/Hooks/TxNewsSitemapGenerator.php [hunk 1]
```diff
@@ -184,6 +184,14 @@ protected function getNewsItemUrl($newsRow, $forceSinglePid = null)
                 'returnLast' => 'url',
                 'useCacheHash' => true,
             ];
+            if ($newsRow['type'] == 1 && !empty($newsRow['internalurl'])) {
```
Comment (georgringer) [inline Classes/Hooks/TxNewsSitemapGenerator.php]: please cast it to integer

File: Classes/Hooks/TxNewsSitemapGenerator.php [hunk 2]
```diff
@@ -184,6 +184,14 @@ protected function getNewsItemUrl($newsRow, $forceSinglePid = null)
...
+                $conf['additionalParams'] = $additionalParams;
+                $conf['parameter'] = $newsRow['internalurl'];
+            }
+            if ($newsRow['type'] == 2 && !empty($newsRow['externalurl'])) {
```
Comment (georgringer) [inline Classes/Hooks/TxNewsSitemapGenerator.php]: please use an elseif

File: Classes/Hooks/TxNewsSitemapGenerator.php (hunk 1 above)
Comment (0bserver) [reply Classes/Hooks/TxNewsSitemapGenerator.php]: Done …

File: Classes/Hooks/TxNewsSitemapGenerator.php (hunk 2 above)
Comment (0bserver) [reply Classes/Hooks/TxNewsSitemapGenerator.php]: … and done. Good point.

**Comment by georgringer** [comment]
fixed with different commit

**Pull request closed without merging**